service/tasks/sync_project.py

```diff
@@ -0,0 +1,199 @@
+    return response.text
+
+
+def get_compute_url(json_catalog):
+    celery_logger.info("start of get_compute_url")
+    try:
+        catalog = json.loads(json_catalog)
+    except KeyError:
+        celery_logger.info("Invalid token passed")
+        raise serializers.ValidationError("Invalid token passed")
+    endpoint_catalog = catalog['token']['catalog']
+    compute = None
+    for item in endpoint_catalog:
+        if item['type'] == 'compute':
+            compute = item
+    if not compute:
+        celery_logger.info("Cannot find compute endpoint catalog")
+        raise serializers.ValidationError("Cannot find compute endpoint catalog")
+    compute_url = None
+    for ep in compute['endpoints']:
+        if ep['interface'] == 'public':
+            compute_url = ep['url']
+    if not compute_url:
+        celery_logger("Cannot find a public compute enpoint url")
+        raise serializers.ValidationError("Cannot find a public compute endpoint url")
+    celery_logger.info("end of get_compute_url")
+    return compute_url
+
+
+# Removing celery so that we can test this
+# @task(name="sync_atm_with_openstack")
+def sync_atm_with_openstack(identity_id):
+    #  - There is nothing to return here.  It just syncs the database
+    celery_logger.info("sync_atm_with_openstack:  identity_id=%s"
+                       % (str(identity_id)))
+    identity = Identity.objects.get(id=identity_id)
+    atm_user = identity.created_by
+    creds = identity.get_all_credentials()
+    auth_url = creds['ex_force_auth_url'] + '/v3'
+    auth_token = creds['ex_force_auth_token']
+    (unscoped_token, unscoped_sess) = get_unscoped_token_and_session(auth_url,
+                                                                     auth_token)
+    os_user_id = unscoped_sess.get_user_id()
+    unscoped_keystone = client.Client(session=unscoped_sess)
+    os_project_list = unscoped_keystone.projects.list(user=os_user_id)
+    atm_group = get_or_create_user_group(atm_user)
+    # keep track of all of the active projects so it is known which ones to
+    # delete.
+    active_atm_projects = []
+    for os_project in os_project_list:
+        celery_logger.info("Syncing OS Project: %s:%s %s" % (str(os_project.domain_id),
+                           str(os_project.name), str(os_project.id)))
+        atm_project = get_atm_project(os_project, atm_group.id)
+        if atm_project:
+            scoped_token = get_os_scoped_token(auth_url,
+                                               unscoped_token,
+                                               os_project.id)
+            catalog_json = get_os_project_catalog(auth_url, scoped_token)
+            compute_url = get_compute_url(catalog_json)
+            # In the future, we will need to process the catalog to find the urls
+            # to each of the services
+            identity.update_credential(identity,
+                                       'tok_' + str(os_project.id),
```

Comment:
Because project_name is potentially ambiguous whereas the project id is not.  This seems simpler than specifying tok_[project_domain_id]_[project_name].  Furthermore, we can link this id up with the project in the project table (os_project_id).  Having one unique id is easier to deal with than a combination of 2 ids.